components/admin_console/schema_admin_settings.jsx

```diff
@@ -0,0 +1,418 @@
+                    values={setting.label_values}
+                />
+            );
+        }
+        return setting.label;
+    }
+
+    renderHelpText = (setting) => {
+        if (!this.props.schema) {
+            return <span>{''}</span>;
+        }
+
+        if (this.props.schema.translate === false) {
+            return <span>{setting.help_text}</span>;
+        }
+
+        if (typeof setting.help_text === 'string') {
+            if (setting.help_text_html) {
+                return (
+                    <FormattedHTMLMessage
+                        id={setting.help_text}
+                        values={setting.help_text_values}
+                        defaultMessage={setting.help_text_default}
+                    />
+                );
+            }
+            return (
+                <FormattedMessage
+                    id={setting.help_text}
+                    defaultMessage={setting.help_text_default}
+                    values={setting.help_text_values}
+                />
+            );
+        }
+        return setting.help_text;
+    }
+
+    renderLabel = (setting) => {
+        if (!this.props.schema) {
+            return '';
+        }
+
+        if (this.props.schema.translate === false) {
+            return setting.label;
+        }
+        return Utils.localizeMessage(setting.label, setting.label_default);
+    }
+
+    isDisabled = (setting) => {
+        if (setting.needs) {
+            for (const need of setting.needs) {
+                if (this.state[need[0]] !== need[1]) {
+                    return true;
+                }
+            }
+        }
+        if (setting.needs_license && !this.props.license.IsLicensed) {
+            return true;
+        }
+        if (setting.needs_no_license && this.props.license.IsLicensed) {
+            return true;
+        }
+        return false;
+    }
+
+    buildSetting = (id, setting) => {
+        switch (setting.type) {
+        case SettingsTypes.TYPE_TEXT:
+            return this.buildTextSetting(id, setting);
+        case SettingsTypes.TYPE_NUMBER:
+            return this.buildTextSetting(id, setting);
+        case SettingsTypes.TYPE_BOOL:
+            return this.buildBoolSetting(id, setting);
+        case SettingsTypes.TYPE_DROPDOWN:
+            return this.buildDropdownSetting(id, setting);
+        case SettingsTypes.TYPE_RADIO:
+            return this.buildRadioSetting(id, setting);
+        case SettingsTypes.TYPE_BANNER:
+            return this.buildBannerSetting(id, setting);
+        case SettingsTypes.TYPE_GENERATED:
+            return this.buildGeneratedSetting(id, setting);
+        case SettingsTypes.TYPE_USERNAME:
+            return this.buildUsernameSetting(id, setting);
+        case SettingsTypes.TYPE_BUTTON:
+            return this.buildButtonSetting(id, setting);
+        case SettingsTypes.TYPE_LANGUAGE:
+            return this.buildLanguageSetting(id, setting);
+        }
```

Comment:
Feels like repetitive to me. Why not like?
```
buildSetting = {
    [SettingsTypes.TYPE_TEXT]: this.buildTextSetting,
    [SettingsTypes.TYPE_NUMBER]: this.buildTextSetting,
    ...
}
```
then maybe call it like:
```
buildSetting[setting.type](setting.key, setting)
```

Reply:
Yes, probably something like `buildSetting[setting.type] && buildSetting[setting.type](setting.key, setting)`, because you can have an invalid type. But I like a lot your approach :)